ray-operator/controllers/ray/common/service.go

```diff
@@ -276,6 +276,39 @@ func BuildServeService(ctx context.Context, rayService rayv1.RayService, rayClus
 	return serveService, nil
 }
 
+// BuildHeadlessService builds the headless service for workers in multi-host worker groups to communicate
+func BuildHeadlessServiceForRayCluster(rayCluster rayv1.RayCluster) (*corev1.Service, error) {
+	name := rayCluster.Name + utils.DashSymbol + utils.HeadlessServiceSuffix
+	namespace := rayCluster.Namespace
+
+	labels := map[string]string{
+		utils.RayClusterHeadlessServiceLabelKey: rayCluster.Name,
+	}
+	selectorLabels := map[string]string{
+		utils.RayClusterLabelKey:  rayCluster.Name,
+		utils.RayNodeTypeLabelKey: string(rayv1.WorkerNode),
```

Comment:
I want to confirm: If I have 4 worker groups, including 1 group for CPU, 1 for GPU, 1 for single-host TPU, and 1 for multi-host TPU, is it the expected behavior to attach all Pods from all these groups to the headless service?

Reply:
Yeah that's expected behavior, since I don't believe there are any existing labels assigned to the pods that would allow us to filter only the multi-host TPUs. As long as the multi-host TPUs are exposed by the headless service this should work for the JAX initialization.